File: src/main/java/com/lessmarkup/interfaces/cache/AbstractCacheHandler.java
package com.lessmarkup.interfaces.cache;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collection;
import java.util.OptionalLong;

public abstract class AbstractCacheHandler implements CacheHandler {

    private final Collection<Class<?>> handledCollectionTypes;
    
    protected AbstractCacheHandler(Class<?>[] handledCollectionTypes)
    {
        this.handledCollectionTypes = new ArrayList<>();    
        if (handledCollectionTypes != null) {
            Arrays.stream(handledCollectionTypes).forEach(this.handledCollectionTypes::add);
        }
    }
    
    @Override
    public void initialize(OptionalLong objectId) {
        if (objectId.isPresent()) {
            throw new IllegalArgumentException();
        }
    }

    @Override
    public boolean expires(int collectionId, long entityId, EntityChangeType changeType) {
        return true;
    }

    @Override
    public Collection<Class<?>> getHandledCollectionTypes() {
        return handledCollectionTypes;
    }

    @Override
    public boolean isExpired() {
        return false;
    }
    
}

File: src/main/scala/com/lessmarkup/interfaces/cache/AbstractCacheHandler.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.interfaces.cache

abstract class AbstractCacheHandler(val handledCollectionTypes: Seq[Class[_]] = Nil) extends CacheHandler {

  def initialize(objectId: Option[Long]) {
    if (objectId.isDefined) {
      throw new IllegalArgumentException
    }
  }

  def expires(collectionId: Int, entityId: Long, changeType: EntityChangeType) = true

  def isExpired: Boolean = false
}